components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -125,17 +139,67 @@ void Nudging::initialize_impl (const RunType /* run_type */)
 {
   using namespace ShortFieldTagsNames;
 
-  // Initialize the time interpolator
-  auto grid_ext = m_grid->clone(m_grid->name(), false);
+  // Initialize the refining remapper stuff at the outset,
+  // because we need to know the grid information
+  if(m_refine_remap) {
+    // For now, we are doing the horizontal interpolation last,
+    // so we use the m_grid (model physics) as the target
+    // TODO: maybe clean this up?
+    auto grid_tgt = m_grid->clone(m_grid->name(), false);
+    auto refine_remapper_p2p =
+        std::make_shared<RefiningRemapperP2P>(grid_tgt, m_refine_remap_file);
+    refine_remapper = refine_remapper_p2p;
```

Comment:
IF you ever get bored of doing lots of `if (refine_remapper)`, you could consider _always_ have a remapper, with the catch that, if remap is not needed, it is of type [DoNothingRemapper](https://github.com/E3SM-Project/scream/blob/master/components/eamxx/src/share/grid/remap/do_nothing_remapper.hpp), which is kind of a placeholder. But that way, you can _always_  call stuff like `refine_remapper->registration_ends();` (it's a no-op for `DoNothingRemapper`). You can even call the remap methods (which are also no-op methods), or use it to get the ext grid (which would be the remapper src grid).

This is really not necessary, but I'm pointing it out in case you notice that all these if statements make the code less readable.